python/xgrammar/compiler.py

```diff
@@ -48,18 +53,37 @@ class GrammarCompiler(XGRObject):
 
     cache_enabled : bool, default: True
         Whether to enable the cache.
+
+    max_memory_usage : float, default: 1024
+        The maximum memory usage for the cache in the specified unit.
+        Note that the actual memory usage may slightly exceed this value.
+
+    memory_unit : Literal["B", "KiB", "MiB", "GiB"], default: "MiB"
+        The unit of the memory usage.
```

Comment:
```suggestion
    cache_limit_bytes : int, default: -1
        The maximum memory usage for the cache in the specified unit.
        Note that the actual memory usage may slightly exceed this value.
```
don't need memory unit here to keep it simple